### PR TITLE
[Fix] JS: Empty group

### PR DESCRIPTION
**Pull request by rldhont**
Empty group is view as layer in tree config, but it is an empty group. So for empty group no layer state has to be build.

Funded by Grand Narbonne

### CODE REVIEW

**Review by mdouchin — APPROVED** [review]

File: tests/qgis-projects/tests/display_in_legend.qgs.cfg [hunk 1]
```diff
@@ -1,14 +1,17 @@
 {
     "metadata": {
-        "qgis_desktop_version": 32216,
-        "lizmap_plugin_version_str": "4.1.8",
-        "lizmap_plugin_version": 40108,
+        "qgis_desktop_version": 32815,
+        "lizmap_plugin_version_str": "4.3.3",
```
Comment (Gustry) [inline tests/qgis-projects/tests/display_in_legend.qgs.cfg:4]: You should upgrade your plugin version ;-)

**Review by Gustry — APPROVED** [review]

File: tests/js-units/data/display-in-legend-capabilities.json [hunk 2]
```diff
@@ -357,6 +357,70 @@
                     "opaque": false,
                     "noSubsets": false
                 },
+                {
+                    "Name": "group-without-children",
+                    "Title": "group-without-children",
+                    "CRS": [
+                        "CRS:84",
```
Comment (nboisteault) [inline tests/js-units/data/display-in-legend-capabilities.json:364]: Why repeat?

File: tests/qgis-projects/tests/treeview.qgs.cfg [hunk 3]
```diff
@@ -1,16 +1,15 @@
 {
     "metadata": {
         "qgis_desktop_version": 32815,
-        "lizmap_plugin_version_str": "4.3.5-alpha",
-        "lizmap_plugin_version": 40305,
-        "lizmap_web_client_target_version": 30800,
-        "lizmap_web_client_target_status": "Dev",
-        "instance_target_url": "http://localhost:8130/",
-        "instance_target_repository": "testsrepository"
+        "lizmap_plugin_version_str": "4.3.3",
```
Comment (nboisteault) [inline tests/qgis-projects/tests/treeview.qgs.cfg:4]: You should update your plugin.

Comment (Gustry) [reply tests/qgis-projects/tests/treeview.qgs.cfg:4]: https://github.com/3liz/lizmap-web-client/pull/4426#discussion_r1599479717

File: tests/js-units/data/display-in-legend-capabilities.json (hunk 2 above)
Comment (rldhont) [reply tests/js-units/data/display-in-legend-capabilities.json:364]: I don't know it is generated by OpenLayers from the WMS GetCapabilities

**Pull request merged**